app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/GoogleSignInWidget.kt

```diff
@@ -94,16 +137,50 @@ class GoogleSignInWidget : Fragment() {
         }
     }
 
+    private fun setSingInUi() {
+        view?.findViewById<Button>(R.id.singOutButton)?.visibility = View.GONE
+        view?.findViewById<Button>(R.id.checkBox)?.visibility = View.VISIBLE
+        view?.findViewById<Button>(R.id.loginButton)?.visibility = View.VISIBLE
+    }
+
+    private fun setSingOutUi() {
+        view?.findViewById<Button>(R.id.singOutButton)?.visibility = View.VISIBLE
+        view?.findViewById<Button>(R.id.checkBox)?.visibility = View.GONE
+        view?.findViewById<Button>(R.id.loginButton)?.visibility = View.GONE
 
-    private fun updateUi() {
-        val user = presenter.currentUser
-        view?.findViewById<TextView>(R.id.loginStatus)?.text =
-            user?.let {
-                getString(
-                    R.string.SignInFragment_loginStatusSuccessMessage,
-                    user.displayName
-                )
-            } ?: "Not logged in!"
+    }
+
+
+    fun updateUi() {
+        var user: User
+        if (prefs!!.getBoolean("hasLogin", false)) {
+            view?.findViewById<TextView>(R.id.loginStatus)?.text =
+                prefs?.getString("name", "Not logged in!")
+            user = User(
+                prefs!!.getString("name", ""),
+                prefs!!.getString("email", ""),
+                prefs!!.getString("uid", "")!!,
+                User.Type.FIREBASE,
+                prefs!!.getBoolean("isAdmin", false)
+            )
+            presenter.currentUser = user
```

Comment:
It's not the role of the view to tell the presenter what the current user is. This is incorrectly designed.

Reply:
Yes Agreed also this piece of code is unnecessary after using the shared preferences I will remove it